sweep/sweeper.go

```diff
@@ -676,11 +716,22 @@ func (s *UtxoSweeper) removeExclusiveGroup(group uint64) {
 			continue
 		}
 
+		// Skip inputs that are already terminated.
+		if input.terminated() {
+			log.Tracef("Skipped sending error result for "+
+				"input %v, state=%v", outpoint, input.state)
+
+			continue
+		}
+
 		// Signal result channels.
-		s.signalAndRemove(&outpoint, Result{
+		s.signalResult(input, Result{
 			Err: ErrExclusiveGroupSpend,
 		})
 
+		// Update the input's state as it can no longer be swept.
+		input.state = StateExcluded
```

Comment:
Will this actually modify the value in the map, or do we also need to do a re-assignment?

Reply:
Looks to be fine: https://go.dev/play/p/W06zPXRwDFS

Reply:
yeah `input` is a pointer so we are fine here.